src/Symfony/Component/Validator/Constraints/BackedEnumValueValidator.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\ConstraintDefinitionException;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+
+/**
+ * EnumValidator validates that the value is one of the expected values.
+ *
+ * @author Aurélien Pillevesse <aurelienpillevesse@hotmail.fr>
+ */
+class BackedEnumValueValidator extends ConstraintValidator
+{
+    public function validate(mixed $value, Constraint $constraint): void
+    {
+        if (!$constraint instanceof BackedEnumValue) {
+            throw new UnexpectedTypeException($constraint, BackedEnumValue::class);
+        }
+
+        if (!is_a($constraint->type, \BackedEnum::class, true)) {
+            throw new ConstraintDefinitionException('The "type" must be a \BackedEnum.');
+        }
+
+        if (null === $value || '' === $value) {
+            return;
+        }
+
+        $backingType = (string) (new \ReflectionEnum($constraint->type))->getBackingType();
+        if ($this->getNormalizedType($value) !== $backingType) {
+            $this->context->buildViolation($constraint->typeMessage)
+                ->setParameter('{{ type }}', $this->formatValue($backingType))
+                ->setCode(BackedEnumValue::INVALID_TYPE_ERROR)
+                ->addViolation();
+        }
+
+        try {
+            $enumTypeValue = $constraint->type::tryFrom($value);
+        } catch (\TypeError) {
+            $cases = $constraint->type::cases();
+            if (\count($cases) > 0) {
+                $this->context->buildViolation($constraint->message)
+                    ->setParameter('{{ value }}', $this->formatValue($value))
+                    ->setParameter('{{ choices }}', $this->formatValues(array_map(fn ($case) => $case->value, $cases)))
+                    ->setCode(BackedEnumValue::NO_SUCH_CHOICE_ERROR)
+                    ->addViolation();
+            }
+
+            return;
+        }
```

Comment:
By checking against the backing type first, you have effectively ruled out the possibility of `TypeError`s. This catch is dead code.